django/utils/url_config.py

```diff
@@ -0,0 +1,135 @@
+import inspect
+from collections import defaultdict, namedtuple
+from urllib import parse
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import module_loading
+
+BACKENDS = defaultdict(dict)
+
+BaseParsedURL = namedtuple('ParsedURL', 'scheme username password hostname port path options')
```

Comment:
Could you inherit from [`urllib.parse.ParseResult`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.ParseResult) to make the class hierarchy easier for people already familiar with the stdlib's data types? Then you could do away with the `namedtuple` import and simply do
```python
from urllib import parse
class ParseResult(parse.ParseResult):
    @property
    def netloc(self):
        # ...what you had written below...
```
This should override `ParseResult.netloc` correctly.

Reply:
Great idea! 👍

Reply:
Ahh yes. I remember why I did all this, it's because really the ParseResult is pretty unsuited for this due to a lot of annoying quirks. I think it should be a class entirely distinct from ParseResult.